File: src/js/_0160_Intersection_of_Two_Linked_Lists.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} headA
 * @param {ListNode} headB
 * @return {ListNode}
 */
var getIntersectionNode = function (headA, headB) {
    if (headA === null || headB === null)
        return null;

    let a = headA, b = headB;

    while (a !== b) {
        a = a !== null ? a.next : headB;
        b = b !== null ? b.next : headA;
    }

    return a;
};

File: src/_0160_Intersection_of_Two_Linked_Lists.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode *getIntersectionNode(struct ListNode *headA, struct ListNode *headB)
{
    if (headA == NULL || headB == NULL)
        return NULL;

    struct ListNode *a = headA, *b = headB;

    while (a != b)
    {
        a = a ? a->next : headB;
        b = b ? b->next : headA;
    }

    return a;
}
